R/zchunk_L222.en_transformation_USA.R

```diff
@@ -67,210 +75,638 @@ module_gcam.usa_L222.en_transformation_USA_DISABLED <- function(command, ...) {
+          rename(market.name = grid_region) ->
+          tmp
+
+        L222.StubTechMarket_en_USA %>%
+          filter(!(minicam.energy.input %in% gcamusa.REGIONAL_FUEL_MARKETS)) %>%
+          bind_rows(tmp) ->
+          L222.StubTechMarket_en_USA
+      }
+
+
+      # Finish L222.StubTechMarket_en_USA by Setting electricity to the state markets
+      L222.StubTechMarket_en_USA %>%
+        filter(minicam.energy.input %in% gcamusa.ELECT_TD_SECTORS) %>%
+        mutate(market.name = region) ->
+        tmp
+
+      # create a key for filtering
+      L222.StubTech_en_USA %>%
+        select(supplysector, subsector, stub.technology) %>%
+        unite(key, supplysector, subsector, stub.technology, sep = "~") %>%
+        distinct ->
+        L222.StubTech_en_USA_key
+
+      L222.StubTechMarket_en_USA %>%
+        filter(!(minicam.energy.input %in% gcamusa.ELECT_TD_SECTORS)) %>%
+        bind_rows(tmp) %>%
+        select(one_of(LEVEL2_DATA_NAMES[["StubTechMarket"]])) %>%
+        unite(key, supplysector, subsector, stub.technology, sep = "~") %>%
+        filter(key %in% L222.StubTech_en_USA_key$key) %>%
+        separate(key, c("supplysector", "subsector", "stub.technology"), sep = "~") %>%
+        filter((subsector == "oil refining" & region %in% oil_refining_states) |
+                 subsector != "oil refining") ->
+        L222.StubTechMarket_en_USA
+
+
+      # L222.CarbonCoef_en_USA: energy carbon coefficients in USA
+      #
+      # Step 1, process L202.CarbonCoef for joining
+      L202.CarbonCoef %>%
+        filter(region == "USA") %>%
+        select(-region) %>%
+        distinct ->
+        L202.CarbonCoef_tmp
+
+      # Step 2, create L222.CarbonCoef_en_USA by joining the table from step 1.
+      L222.Supplysector_en_USA %>%
+        select(region, supplysector) %>%
+        distinct %>%
+        left_join_error_no_match(distinct(select(L222.TechShrwt_USAen, subsector, supplysector)),
+                                 by = c("supplysector" = "subsector")) %>%
+        rename(PrimaryFuelCO2Coef.name = supplysector.y) %>%
+        left_join_error_no_match(L202.CarbonCoef_tmp, by =  "PrimaryFuelCO2Coef.name") %>%
+        select(-supplysector) ->
+        L222.CarbonCoef_en_USA
+
+
+
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # L222.DeleteStubTech_USAen %>%
+      tibble() %>%
+      add_title("Removes existing stub technologies in the USA region") %>%
```

Comment:
Odd title--can it be clearer? E.g., is this a table of technologies that GCAM will delete in the USA?